src/Common/src/TypeSystem/Common/TargetDetails.cs

```diff
@@ -115,11 +115,28 @@ public bool SupportsRelativePointers
         /// <summary>
         /// Gets the maximum alignment to which something can be aligned
         /// </summary>
-        public static int MaximumAlignment
+        public int MaximumAlignment
```

Comment:
Looking at this now, maybe `MaximumAlignment` should actually be moved to the type system context if we decide to go the instance field route.

Reply:
MaximumAlignment is an optimization for universal shared generic code. The moment it becomes too big like what we are doing here it becomes useless. We can just drop it completely.